unittests/sync_call_tests.cpp

```diff
@@ -1295,16 +1183,23 @@ static const char direct_recursive_wast[] = R"=====(
 BOOST_AUTO_TEST_CASE(direct_recursive_depth_enforcement_test)  { try {
    validating_tester t;
 
-   if( t.get_config().wasm_runtime == wasm_interface::vm_type::eos_vm_oc ) {
-      // skip eos_vm_oc for now.
-      return;
-   }
-
    create_accounts_and_set_code(one_input_caller_wast, direct_recursive_wast, t);
 
    // Do a recursive call with n == 1
    BOOST_REQUIRE_NO_THROW(t.push_action("caller"_n, "callwithinpt"_n, "caller"_n, mvo()("input", "1")));
 
+   if( t.get_config().wasm_runtime == wasm_interface::vm_type::eos_vm_oc ) {
+      // Do a small recursive call for now
+      BOOST_REQUIRE_NO_THROW(t.push_action("caller"_n, "callwithinpt"_n, "caller"_n, mvo()("input", 3)));
+      return;
+   }
+
+   if( t.get_config().wasm_runtime == wasm_interface::vm_type::eos_vm_oc ) {
+      // Do a small recursive call for now
+      BOOST_REQUIRE_NO_THROW(t.push_action("caller"_n, "callwithinpt"_n, "caller"_n, mvo()("input", 3)));
+      return;
+   }
+
```

Comment:
Need to get the memory situation handled before ready for full review.

Currently the main thread memory is created with full 528 slices, and the RO thread memories are created with 10 slices.

For the simplest approach to get it floating, I'd suggest maybe just making every sync call use memory with a single slice.

Ideally we do something more complex where shallow sync calls get more slices than deeper sync calls (under the premise that shallow calls are more frequent). But we'd need to work out the math on this first.

Reply:
Done with a single slice for now. Created an issue https://github.com/AntelopeIO/spring/issues/1393.